src/platform/macos.rs

```diff
@@ -55,6 +55,22 @@ pub trait WindowExtMacOS {
 
     /// Put the window in a state which indicates a file save is required.
     fn set_document_edited(&self, edited: bool);
+
+    /// Toggles the `Option` key being interpretted as an `Alt` modifier.
+    ///
+    /// This will ignore diacritical marks and accent characters from
+    /// being processed as received characters. Instead, the input
+    /// device's raw character will be placed in event queues with the
+    /// Alt modifier set.
```

Comment:
Reading this, I'm still unsure of what the intended effect is? Maybe you could add a few examples of the desired behaviour, something like this:

```rust
// Press OPTION/ALT key
KeyboardInput { state: Pressed, keycode: Some(Alt), ... }
ModifiersChanged(ALT)

// Press E key
match (ime_allowed, option_as_alt) {
    (true, false) => {
        Ime::Preedit("´", Some((0, 0)))
    }
    (true, true) | (false, true) => {
        KeyboardInput { state: Pressed, keycode: Some(E), .. }
        ReceivedCharacter('e')
        KeyboardInput { state: Released, keycode: Some(E), .. }
    }
    (false, false) => {
        KeyboardInput { state: Pressed, keycode: Some(E), .. }
        KeyboardInput { state: Released, keycode: Some(E), .. }
    }
}

// Release OPTION/ALT key
KeyboardInput { state: Released, keycode: Some(Alt), ... }
ModifiersChanged()

// Press E key again
if ime_allowed && !option_as_alt {
    Ime::Preedit("", None)
    Ime::Commit("é")
} else {
    KeyboardInput { state: Pressed, keycode: Some(E), .. }
    ReceivedCharacter('e')
    KeyboardInput { state: Released, keycode: Some(E), .. }
}
```

(And the same for an example involving e.g. Pinyin)

Reply:
How about the use case and where I have utilized the fix... 

Basically, when enabled, option should NEVER send an extended character, giving the same behavior as the ALT key on other operating systems. I imagine this feature is really only critical in apps like Alacritty, but it is huge for Alacritty. 

I picked up this issue because I use the combination Alacritty and Zellij on linux. I would like to use the same combination on my mac work machine. Zellij is pretty unusable on Alacritty on the mac due to the option key's default behavior of sending extended characters versus just acting as an ALT key. This actually causes a pretty poor experience in a wide range of terminal apps -- anything that requires an ALT key. 

Every other terminal on the mac (Terminal.app, Iterm.app, Kitty) have the ability to disable sending extended characters, like this fix implements. 

I have been using this fix daily for a while now. If you would like try it, my fork of Alacritty where I have utilized this fix is here: https://github.com/ayax79/alacritty/tree/option_as_alt

Here is my alacritty config: https://github.com/ayax79/config/blob/main/alacritty/alacritty.yml
The relevant section is ```option_as_alt: true``` under ```window:```

When enabled if you run zellij, OPT+n should open a new terminal pane. When disabled OPT+n should give you ˜. 

This has been an outstanding issue in alacritty for a long time: https://github.com/alacritty/alacritty/issues/62